crates/pallets/starknet/src/lib.rs

```diff
@@ -721,8 +721,10 @@ pub mod pallet {
             let transaction_nonce = transaction.nonce;
             let sender_address = transaction.sender_address;
 
+            let nonce_for_priority: u64 = transaction_nonce.try_into().map_err(|_| InvalidTransaction::Future)?;
+
```

Comment:
I'd prefer `InvalidTransaction::Custom` here as it's quite specific to our wrapper and the transaction will NOT be valid in the future.

Reply:
i used `Future` because i saw mention of nonce too high https://crates.parity.io/sp_runtime/transaction_validity/enum.InvalidTransaction.html#variant.Future.

for `Custom` what should be the value of u8?

Reply:
Right but `Future` implies that transaction would be valid in the future which is not right here.

You can do this for `Custom` : `pub(crate) const NONCE_DECODE_FAILURE: u8 = 1;`